src/avkit.cs

```diff
@@ -1099,4 +1100,36 @@ interface AVPlaybackSpeed {
 		[Export ("localizedNumericName")]
 		string LocalizedNumericName { get; }
 	}
+
+	[iOS (17, 2), NoMac, NoMacCatalyst, NoTV, NoWatch]
+	[BaseType(typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface AVCaptureEvent
+	{
+		[Export("phase")]
+		AVCaptureEventPhase Phase { get; }
+	}
+
+	[iOS (17, 2), NoMac, NoMacCatalyst, NoTV, NoWatch]
+	[Native]
+	public enum AVCaptureEventPhase : ulong {
+		Began,
+		Ended,
+		Cancelled,
+	}
+
+	[iOS (17, 2), NoMac, NoMacCatalyst, NoTV, NoWatch]
+	[BaseType(typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface AVCaptureEventInteraction : UIInteraction
+	{
+		[Export ("initWithEventHandler:")]
+		NativeHandle Constructor (Action<AVCaptureEvent> handler);
+
+		[Export ("initWithPrimaryEventHandler:secondaryEventHandler:")]
+		NativeHandle Constructor (Action<AVCaptureEvent> primaryHandler, Action<AVCaptureEvent> secondaryHandler);
```

Comment:
I noticed some places use C# `delegates` for callbacks, and others just use `Actions`. I wasn't sure which is preferred so I went with actions, but I can change these to delegates if that's preferred.

Reply:
The problem with `Action/Func` is that it's not possible to give them names of the parameters, so they have to be pretty much self-explanatory. This means that we typically use delegates when we need to give names to the parameters in order to document what they're doing. In this case, it seems that a delegate is appropriate, since it's not obvious from only the parameter types what each does.

Reply:
In this case, the difference is that the first constructor results in both "primary" and "secondary" events being passed to a single callback, whereas the second constructor splits those events between two callbacks. (Primary and secondary basically just maps to volume down and volume up.)

I was originally planning on using a single delegate such as:
```csharp
delegate void AVCaptureEventHandler(AVCaptureEvent captureEvent);
```

It sounds like maybe you're suggesting three delegates though, as opposed to relying on only constructor parameter names? If so, we could do something like the following. (And perhaps with a better name for the first one.)
```csharp
delegate void AVCaptureEventHandler(AVCaptureEvent captureEvent);
delegate void AVCapturePrimaryEventHandler(AVCaptureEvent primaryEvent);
delegate void AVCaptureSecondaryEventHandler(AVCaptureEvent secondaryEvent);
```

For what it's worth, these are the original methods from the AVKit headers:
```objective-c
- (instancetype)initWithEventHandler:(void (^)(AVCaptureEvent *event))handler

- (instancetype)initWithPrimaryEventHandler:(void (^)(AVCaptureEvent *event))primaryHandler
                      secondaryEventHandler:(void (^)(AVCaptureEvent *event))secondaryHandler
```

Reply:
Ah, never mind, it was too early in the morning when I commented, so I misunderstood the code a bit.

Delegates with a single argument are typically fine as Func/Action, and this doesn't look like an exception.